### PR TITLE
Shorten loc messages

### PR DESCRIPTION
**Pull request by paulvanbrenk**

### CODE REVIEW

File: src/compiler/core.ts [hunk 1]
```diff
@@ -365,10 +365,10 @@ namespace ts {
 
     export let localizedDiagnosticMessages: Map<string> = undefined;
 
-    export function getLocaleSpecificMessage(message: string) {
-        return localizedDiagnosticMessages && localizedDiagnosticMessages[message]
-            ? localizedDiagnosticMessages[message]
-            : message;
+    export function getLocaleSpecificMessage(message: DiagnosticMessage) {
+        return localizedDiagnosticMessages && localizedDiagnosticMessages[message.key]
```
Comment (DanielRosenwasser) [inline src/compiler/core.ts]: For a message that is an empty string, you'll always use the non-localized message unless you use `message.key in localizedDiagnosticMessages`. Though, that would be the most ridiculous edge case I may have ever brought up.

**Comment by DanielRosenwasser** [comment]
:+1:

File: scripts/processDiagnosticMessages.ts [hunk 2]
```diff
@@ -18,6 +18,13 @@ function main(): void {
         return;
     }
 
+    function writeFile(fileName: string, contents: string) {
+        // TODO: Fix path joining
```
Comment (yuit) [inline scripts/processDiagnosticMessages.ts]: why this is a TODO instead of fixing it in the PR

Comment (paulvanbrenk) [inline scripts/processDiagnosticMessages.ts]: because it works ... not sure what @DanielRosenwasser Daniel meant here

Comment (DanielRosenwasser) [inline scripts/processDiagnosticMessages.ts]: https://nodejs.org/api/path.html#path_path_join_path1_path2

**Comment by mhegazy** [comment]
@paulvanbrenk and @DanielRosenwasser is this ready?

**Comment by paulvanbrenk** [comment]
@mhegazy testing and I needed to finish setting up loc support before I could actually check this in. EOD tomorrow.

**Pull request merged**